lib/SimplePie/SimplePie.php

```diff
@@ -1382,6 +1382,43 @@ function cleanMd5($rss)
 		return '';
 	}
 
+	/**
+	 * Removes invalid utf-8 characters.
+	 *
+	 * @access public
+	 * @param string $value
+	 * @return string all valid utf-8 characters
+	 */
+	public static function stripInvalidUtf8($value)
+	{
+		$ret = "";
+		if (empty($value))
+		{
+			return $ret;
+		}
+
+		$length = strlen($value);
+		for ($i=0; $i < $length; $i++)
+		{
+			$current = ord($value[$i]);
+			if (($current == 0x9) ||
+				($current == 0xA) ||
+				($current == 0xD) ||
+				(($current >= 0x20) && ($current <= 0xD7FF)) ||
+				(($current >= 0xE000) && ($current <= 0xFFFD)) ||
+				(($current >= 0x10000) && ($current <= 0x10FFFF)))
+			{
+				$ret .= chr($current);
+			}
+			else
+			{
+				$ret .= " ";
```

Comment:
Or `REPLACEMENT CHARACTER`? I realize it would look uglier, but it's not necessarily clear that hiding the crime would be better.

https://www.fileformat.info/info/unicode/char/fffd/index.htm